config/envoyconfig/listeners.go

```diff
@@ -341,7 +342,7 @@ func (b *Builder) buildMainHTTPConnectionManagerFilter(
 		LuaFilter(luascripts.CleanUpstream),
 		LuaFilter(luascripts.RewriteHeaders),
 	}
-	if tlsDomain != "" && tlsDomain != "*" {
+	if tlsDomain != "" && !strings.Contains(tlsDomain, "*") {
```

Comment:
This function is rather complicated to follow. A comment here (and above?) would be helpful.